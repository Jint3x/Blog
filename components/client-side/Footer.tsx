import React from "react"


function Footer() {
    return (
        <footer className={"w-full bg-black"}>
            <div className={"pt-4 pb-4 flex w-56 justify-between mr-auto ml-auto sm:w-72"}>
                <div>
                    <h3 className={"text-main-green text-base"}>About me</h3>

                    <FooterLink 
                     href={"https://devjint3x.com"}
                     text={"My portfolio"}
                    />

                    <FooterLink 
                     href={"https://www.linkedin.com/in/alexandar-dimitrov-546782216/"}
                     text={"Linkedin"}
                    />  
                </div>
                <div>
                    <h3 className={"text-main-green text-base"}>Find me</h3>

                    <FooterLink 
                     href={"https://github.com/Jint3x"}
                     text={"Github"}
                    />

                    <FooterLink 
                     href={"https://twitter.com/Jint3x"}
                     text={"Twitter"}
                    />
                </div>
            </div>
        </footer>
    )
}


interface FooterLink_I {
    href: string,
    text: string,
}

function FooterLink(props: FooterLink_I) {
    return (
        <a 
         href={props.href}
         className={"text-main-blue text-sm no-underline pt-1 pb-1 block"}
        >
            {props.text}
        </a>
    )
}


export default Footer