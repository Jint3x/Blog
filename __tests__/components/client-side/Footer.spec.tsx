/**
 * @jest-environment jsdom
 */
 import Footer from "../../../components/client-side/Footer";
 import ReactRenderer from "react-test-renderer"
 import { screen, render } from "@testing-library/react"

 test("Validate all links", () => {
    render(<Footer />)

    let headerOne = screen.getByText("About me");
    let portfolio = screen.getByText("My portfolio") as HTMLAnchorElement;
    let linkedin = screen.getByText("Linkedin") as HTMLAnchorElement;

    let headerTwo = screen.getByText("Find me");
    let github = screen.getByText("Github") as HTMLAnchorElement;
    let twitter = screen.getByText("Twitter") as HTMLAnchorElement;

    expect(headerOne.nodeName).toBe("H3");
    expect(portfolio.nodeName).toBe("A");
    expect(portfolio.href).toBe("https://jint3x.dev/");
    expect(linkedin.nodeName).toBe("A");
    expect(linkedin.href).toBe("https://www.linkedin.com/in/jint3x/");

    expect(headerTwo.nodeName).toBe("H3");
    expect(github.nodeName).toBe("A");
    expect(github.href).toBe("https://github.com/Jint3x");
    expect(twitter.nodeName).toBe("A");
    expect(twitter.href).toBe("https://twitter.com/Jint3x");
 })

 test("Take a snapshot of the component", () => {
    let tree = ReactRenderer.create(
        <Footer />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
 })