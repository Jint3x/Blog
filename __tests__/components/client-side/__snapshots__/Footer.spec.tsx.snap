// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot of the component 1`] = `
<footer
  className="w-full bg-black"
>
  <div
    className="pt-4 pb-4 flex w-56 justify-between mr-auto ml-auto sm:w-72"
  >
    <div>
      <h3
        className="text-main-green text-base"
      >
        About me
      </h3>
      <a
        className="text-main-blue text-sm no-underline pt-1 pb-1 block"
        href="https://jint3x.dev"
      >
        My portfolio
      </a>
      <a
        className="text-main-blue text-sm no-underline pt-1 pb-1 block"
        href="https://www.linkedin.com/in/jint3x/"
      >
        Linkedin
      </a>
    </div>
    <div>
      <h3
        className="text-main-green text-base"
      >
        Find me
      </h3>
      <a
        className="text-main-blue text-sm no-underline pt-1 pb-1 block"
        href="https://github.com/Jint3x"
      >
        Github
      </a>
      <a
        className="text-main-blue text-sm no-underline pt-1 pb-1 block"
        href="https://twitter.com/Jint3x"
      >
        Twitter
      </a>
    </div>
  </div>
</footer>
`;
